framework/wazuh/core/configuration.py

```diff
@@ -248,7 +248,14 @@ def _read_option(section_name: str, opt: str) -> tuple:
             if list(opt):
                 for child in opt:
                     child_section, child_config = _read_option(child.tag.lower(), child)
-                    opt_value[child_section] = child_config
+                    if not (section_name, opt_name, child_section) == ('vulnerability-detector', 'provider', 'os'):
+                        opt_value[child_section] = child_config
+                    else:
+                        if opt_value.get(child_section):
+                            opt_value[child_section].append(child_config)
+                        else:
+                            opt_value[child_section] = [child_config]
```

Comment:
This could be reduced using `defaultdict`. It would also increase readability, but I understand that this whole function needs a refactor. We could even consider reading the file as an XML object, but we will discuss it in another issue.

Another option could be:

```suggestion
                        try:
                            opt_value[child_section].append(child_config)
                        except KeyError:
                            opt_value[child_section] = [child_config]
```